### PR TITLE
Run npm registry in Docker

### PR DESCRIPTION
**Pull request by pvdlg**
Run local npm registry for test with [npm-docker-couchdb](https://github.com/npm/npm-docker-couchdb).

Unfortunately it's not possible to run Docker on Travis/OSX or on Appveyor/Windows
See https://github.com/appveyor/ci/issues/1717 and https://github.com/travis-ci/travis-ci/issues/5738

### CODE REVIEW

File: test/helpers/npm-registry.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import Docker from 'dockerode';
+import getStream from 'get-stream';
+import got from 'got';
+import pRetry from 'p-retry';
+
+const SERVER_PORT = 5984;
```
Comment (gr2m) [inline test/helpers/npm-registry.js]: can we change the port? It’s the CouchDB default port, if folks have running it locally, the tests will fail with `Error starting userland proxy: Bind for 0.0.0.0:5984 failed: port is already allocated`

**Comment by gr2m** [comment]
are the tests working for you when run locally? 

I get tons of "unauthorized Name or password is incorrect" errors? Full log: https://gist.github.com/f96f23d4765530db0ad97bae9cbd2500

**Comment by pvdlg** [comment]
The error is exactly what you mentionned [here](https://github.com/semantic-release/semantic-release/pull/530#discussion_r153051451) :)

`Bind for 0.0.0.0:5984 failed: port is already allocated`

File: test/helpers/npm-registry.js (hunk 1 above)
Comment (pvdlg) [reply test/helpers/npm-registry.js]: I made the change.

**Comment by gr2m** [comment]
I’ve stopped my local CouchDB before running the tests. I pulled your latest changes, I still get the same errors. 

I’ve Node 8.9.1 and npm 5.5.1. I’ve deleted node_modules and run `npm install` again. Still the same result. 

To only run the integration tests I tried `./node_modules/.bin/ava  test/integration.test.js`, same results.

Wanna pair on this?

**Comment by gr2m** [comment]
clearing my `~/.npmrc` file made the tests work again

**Pull request merged**